### ISSUE TITLE
Laser in detail view

### ISSUE THREAD

**Issue by KaiKozue** [studio-minus/ppg-bugs-and-requests#965]
I would like to see whenever you turn on detail view to create a laser for the path of the bullet from a gun.

It would only show in detail view and it would show a laser from the barrel of the gun to show where the bullet is going.

![Screenshot 2022-03-23 232007](https://user-images.githubusercontent.com/102275864/159841979-44afefc4-8037-4cf5-953b-9472d4d60c32.png)

**Comment by vda8** [comment]
Guns have ballistics (or "bullet drop" for you dummies out there) and spread, so it won't work.

**Comment by DindinYT37** [comment]
> Guns have ballistics (or "bullet drop" for you dummies out there) and spread, so it won't work.

Then just draw a line with a curve? See no problem with it working, as he only used a laser to show us what he meant and didn't mean to actually use one as that is an in-game element.

**Comment by vda8** [comment]
> > Guns have ballistics (or "bullet drop" for you dummies out there) and spread, so it won't work.
> 
> Then just draw a line with a curve? See no problem with it working, as he only used a laser to show us what he meant and didn't mean to actually use one as that is an in-game element.

What about spread? I guess it would work without tracers. Also, it would take up CPU power. As it has to calculate the trajectory for every frame, and draw a line from it.

**Comment by DindinYT37** [comment]
> > > Guns have ballistics (or "bullet drop" for you dummies out there) and spread, so it won't work.
> > 
> > 
> > Then just draw a line with a curve? See no problem with it working, as he only used a laser to show us what he meant and didn't mean to actually use one as that is an in-game element.
> 
> What about spread? I guess it would work without tracers. Also, it would take up CPU power. As it has to calculate the trajectory for every frame, and draw a line from it.

Not necessarily, if the guns rotation stays the same, you can just move the line along with it - and if the position stays the same, you only have to update it when the rotation changes. Can also be optimized if you don't make the trajectory-line collide.

**Comment by ppgplayer** [comment]
I think vda just does their best to make suggestions not happen.

**Comment by vda8** [comment]
> I think vda just does their best to make suggestions not happen.

I'm just giving my opinion. It would be great for the game. But you can just put a laser on the gun. The suggestion is just not that useful.

**Comment by ppgplayer** [comment]
> > I think vda just does their best to make suggestions not happen.
> 
> I'm just giving my opinion. It would be great for the game. But you can just put a laser on the gun. The suggestion is just not that useful.

Just leave it to zooi. In the end it's their decision.

**Comment by vda8** [comment]
> > > I think vda just does their best to make suggestions not happen.
> > 
> > I'm just giving my opinion. It would be great for the game. But you can just put a laser on the gun. The suggestion is just not that useful.
> 
> Just leave it to zooi. In the end it's their decision.

I'm not trying to decide whether it should be in the game or not. I'm just **commenting**, with the **comment** section.

**Comment by SperoTeStupriMori** [comment]
> I would like to see whenever you turn on detail view to create a laser for the path of the bullet from a gun.
> 
> It would only show in detail view and it would show a laser from the barrel of the gun to show where the bullet is going.
> 
> ![Screenshot 2022-03-23 232007](https://user-images.githubusercontent.com/102275864/159841979-44afefc4-8037-4cf5-953b-9472d4d60c32.png)

no, just no.

**Comment by vda8** [comment]
> > I would like to see whenever you turn on detail view to create a laser for the path of the bullet from a gun.
> > 
> > It would only show in detail view and it would show a laser from the barrel of the gun to show where the bullet is going.
> > 
> > ![Screenshot 2022-03-23 232007](https://user-images.githubusercontent.com/102275864/159841979-44afefc4-8037-4cf5-953b-9472d4d60c32.png)
> 
> no, just no.

Bruh. What's wrong with the suggestion?

**Comment by DindinYT37** [comment]
> > > I would like to see whenever you turn on detail view to create a laser for the path of the bullet from a gun.
> > > It would only show in detail view and it would show a laser from the barrel of the gun to show where the bullet is going.
> > > ![Screenshot 2022-03-23 232007](https://user-images.githubusercontent.com/102275864/159841979-44afefc4-8037-4cf5-953b-9472d4d60c32.png)
> > 
> > 
> > no, just no.
> 
> Bruh. What's wrong with the suggestion?

*Idk he stoopid, source: [953](https://github.com/studio-minus/ppg-bugs-and-requests/issues/953)*

**Comment by KaiKozue** [comment]
I do agree with the attach a laser to the gun I just thought it could be nicer to use in certain situations, but that is just personal preference

**Comment by ppgplayer** [comment]
I think it would be a great idea if greater ballistics were added

**Comment by 404CoderNotFound** [comment]
> > > Guns have ballistics (or "bullet drop" for you dummies out there) and spread, so it won't work.
> > 
> > 
> > Then just draw a line with a curve? See no problem with it working, as he only used a laser to show us what he meant and didn't mean to actually use one as that is an in-game element.
> 
> What about spread? I guess it would work without tracers. Also, it would take up CPU power. As it has to calculate the trajectory for every frame, and draw a line from it.

Make the path a cone instead of a line. Make the curvature of the cone equal the min/max spread of a gun. Boom, done.

**Comment by DindinYT37** [comment]
> > > > Guns have ballistics (or "bullet drop" for you dummies out there) and spread, so it won't work.
> > > 
> > > 
> > > Then just draw a line with a curve? See no problem with it working, as he only used a laser to show us what he meant and didn't mean to actually use one as that is an in-game element.
> > 
> > 
> > What about spread? I guess it would work without tracers. Also, it would take up CPU power. As it has to calculate the trajectory for every frame, and draw a line from it.
> 
> Make the path a cone instead of a line. Make the curvature of the cone equal the min/max spread of a gun. Boom, done.

Yeah but I think guns like the sniper rifle don't have spread, so a line/curved line **and** a cone would be the ideal solution.

**Comment by Gigi-7688** [comment]
Even the sniper rifle has minimal spread after a certain distance (tested)

**Comment by Gigi-7688** [comment]
> Even the sniper rifle has minimal spread after a certain distance (tested)

you can't just calculate the spread at the time of the shot

**Comment by DindinYT37** [comment]
The Idea itself is possible since these kind-of laser pointers are also possible.
![image](https://user-images.githubusercontent.com/64973205/160250014-af3d0663-a0bf-4f0e-9801-55b7ca0a220e.png)

> > Even the sniper rifle has minimal spread after a certain distance (tested)
> 
> you can't just calculate the spread at the time of the shot
I'm pretty sure that the min/max spread is static and only need to be calculated once (preferably on compilation or game start-up, perhaps even when the gun is spawned) as after that, simply making a line that has the correct end-Width (such as the laser pointer from above) should be good enough.
